metric/recorder.go

```diff
@@ -0,0 +1,47 @@
+package metric
+
+import (
+	"context"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/fabric8-services/fabric8-wit/log"
+
+	"github.com/goadesign/goa"
+)
+
+// Recorder record metrics
+func Recorder() goa.Middleware {
+	return func(h goa.Handler) goa.Handler {
+		return func(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+			startTime := time.Now()
+			err := h(ctx, rw, req)
+			recordReqCnt(ctx, req)
+			if err == nil {
```

Comment:
We would like to record requests that fail as well

Reply:
I guess for failed request we need another metric.  Here, We should record only successful request so when we use this metric to analyze system performance, we only considering request which actually serve the purpose.